src/openvic-simulation/politics/NationalFocus.cpp

```diff
@@ -90,40 +93,28 @@ bool NationalFocusManager::load_national_foci_file(
 			std::string_view identifier, ast::NodeCPtr node
 		) -> bool {
 			uint8_t icon = 0;
+			bool has_flashpoint = false, own_provinces = true, outliner_show_as_percent = false;
 			ModifierValue modifiers;
-			NationalFocus::pop_promotion_map_t promotions;
-			NationalFocus::production_map_t production;
 			Ideology const* loyalty_ideology = nullptr;
 			fixed_point_t loyalty_value = 0;
-			ConditionScript limit { scope_t::PROVINCE | scope_t::COUNTRY, scope_t::PROVINCE | scope_t::COUNTRY, scope_t::NO_SCOPE };
+			ConditionScript limit {
+				scope_t::PROVINCE | scope_t::COUNTRY, scope_t::PROVINCE | scope_t::COUNTRY, scope_t::NO_SCOPE
+			};
 
-			bool ret = modifier_manager.expect_modifier_value_and_keys_and_default(
+			bool ret = modifier_manager.expect_modifier_value_and_keys(
 				move_variable_callback(modifiers),
-				[&promotions, &pop_manager, &production, &good_manager, &modifiers, &modifier_manager](
-					std::string_view key, ast::NodeCPtr value
-				) -> bool {
-					PopType const* pop_type = pop_manager.get_pop_type_by_identifier(key);
-					if (pop_type != nullptr) {
-						return expect_fixed_point(map_callback(promotions, pop_type))(value);
-					}
-					Good const* good = good_manager.get_good_by_identifier(key);
-					if (good != nullptr) {
-						return expect_fixed_point(map_callback(production, good))(value);
-					}
-					return key_value_invalid_callback(key, value);
-				},
 				"icon", ONE_EXACTLY, expect_uint(assign_variable_callback(icon)),
 				"ideology", ZERO_OR_ONE,
```

Comment:
ideology & loyalty_value together are also a state modifier.
They add party loyalty over time. Besides that, looks good.

Reply:
Yeah I wanted to make that a `ModifierEffect` too but there's no way to do it within the modifier framework, everything is based on individual keys, whether that has a number as its value or a list of sub-modifiers, but `ideology` and `loyalty_value` are at the same level.